src/Appwrite/Auth/Key.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Appwrite\Auth;
+
+use Ahc\Jwt\JWT;
+use Ahc\Jwt\JWTException;
+use Appwrite\Extend\Exception;
+use Utopia\Config\Config;
+use Utopia\Database\DateTime;
+use Utopia\Database\Document;
+use Utopia\System\System;
+
+class Key
+{
+    public function __construct(
+        protected string $projectId,
+        protected string $type,
+        protected string $role,
+        protected array $scopes,
+        protected string $name,
+        protected bool $expired = false,
+        protected array $disabledMetrics = [],
+    ) {
+    }
+
+    public function getProjectId(): string
+    {
+        return $this->projectId;
+    }
+
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    public function getRole(): string
+    {
+        return $this->role;
+    }
+
+    public function getScopes(): array
+    {
+        return $this->scopes;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function isExpired(): bool
+    {
+        return $this->expired;
+    }
+
+    public function getDisabledMetrics(): array
+    {
+        return $this->disabledMetrics;
+    }
+
+    /**
+     * Decode the given secret key into a Key object, containing the project ID, type, role, scopes, and name.
+     * Can be a stored API key or a dynamic key (JWT).
+     *
+     * @param Document $project
+     * @param string $key
+     * @return Key
+     * @throws Exception
+     */
+    public static function decode(
+        Document $project,
+        string $key
+    ): Key {
+        if (\str_contains($key, '_')) {
+            [$type, $secret] = \explode('_', $key, 2);
+        } else {
+            $type = API_KEY_STANDARD;
+            $secret = $key;
+        }
+
+        $role = Auth::USER_ROLE_APPS;
+        $roles = Config::getParam('roles', []);
+        $scopes = $roles[Auth::USER_ROLE_APPS]['scopes'] ?? [];
```

Comment:
This on root level feels like by default, scopes of app rule will be used.

But it seems to only be used in dynamic key. Can we move it inside it's switch/case?

Reply:
This is used for both key types, it gets merged with the standard key's scopes too